scipy/signal/tests/test_signaltools.py

```diff
@@ -3912,117 +3991,165 @@ def test_nonnumeric_dtypes(func, xp):
         func(*args, x=1.)
 
 
-@skip_xp_backends(np_only=True)
-@pytest.mark.parametrize('dt', 'fdFD')
+@skip_xp_backends(cpu_only=True)
```

Comment:
is it deliberate to be skipping these tests for CuPy too? (apologies if I haven't followed the conversation above)

Reply:
It's deliberate, yes. There a collection of issues: a deliberate incompatibility of `_zi` functions, per Edgar's explanation above, and a collection of cupy bugs to be hopefully fixed when cupy 14 is released. The last commit adds a lengthy code comment.